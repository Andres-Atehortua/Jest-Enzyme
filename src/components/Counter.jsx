import React, { useState } from "react";

const Counter = () => {
  const [count, setIncrement] = useState(0);

  const increment = () => setIncrement(count + 1);
  const decrement = () => setIncrement(count - 1);
  return (
    <div>
      <h2>{count}</h2>
      <button onClick={increment}>Increment</button>
      <button onClick={decrement}>Decrement</button>
    </div>
  );
};

export default Counter;
