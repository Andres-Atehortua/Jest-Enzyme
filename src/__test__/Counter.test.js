import React from "react";
import Counter from "./../components/Counter";
import { configure, shallow /* mount, render*/ } from "enzyme";
import toJson from "enzyme-to-json"; // Dar otro formato al snapshot
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("Pruebas para componente <Counter />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Counter />);
  });

  test("Comprobar que coincida con el snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("Validar funcionamiento de botones", () => {
    wrapper.find("button").first().simulate("click");
    expect(wrapper.find("h2").text()).toBe("1");
    wrapper.find("button").last().simulate("click");
    wrapper.find("button").last().simulate("click");
    wrapper.find("button").last().simulate("click");
    expect(wrapper.find("h2").text()).toBe("-2");
  });
});
