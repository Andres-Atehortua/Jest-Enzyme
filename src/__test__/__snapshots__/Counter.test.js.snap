// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para componente <Counter /> Comprobar que coincida con el snapshot 1`] = `
<div>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
